### ISSUE TITLE
[flaky CI] NodeNetworkConfigurationPolicy bonding default interface when there is a default interface with dynamic address should successfully move default IP address on top of the bond 

### ISSUE THREAD

**Issue by qinqon** [nmstate/kubernetes-nmstate#569]
**What happened**:
The following test is failing at NOOP PR https://github.com/nmstate/kubernetes-nmstate/pull/568:
`NodeNetworkConfigurationPolicy bonding default interface when there is a default interface with dynamic address should successfully move default IP address on top of the bond `

https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/568/pull-kubernetes-nmstate-e2e-handler-k8s/1276499359113940992

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- `NodeNetworkState` on affected nodes (use `kubectl get nodenetworkstate <node_name> -o yaml`):
- Problematic `NodeNetworkConfigurationPolicy`:
- kubernetes-nmstate image (use `kubectl get pods --all-namespaces -l app=kubernetes-nmstate -o jsonpath='{.items[0].spec.containers[0].image}'`):
- NetworkManager version (use `nmcli --version`)
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Others:

**Comment by akrejcir** [comment]
Happened again on 28.10.2020
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/610/pull-kubernetes-nmstate-e2e-handler-k8s/1321402949242458113

The error message is different:
```
/tmp/knmstate/kubernetes-nmstate/test/e2e/handler/bonding_default_interface_test.go:49
Timed out after 480.000s.
Expected
    <shared.ConditionList | len:2, cap:2>: [
        {
            Type: "Available",
            Status: "False",
            Reason: "",
            Message: "",
            LastHeartbeatTime: {
                Time: 0001-01-01T00:00:00Z,
            },
            LastTransitionTime: {
                Time: 0001-01-01T00:00:00Z,
            },
        },
        {
            Type: "Degraded",
            Status: "True",
            Reason: "",
            Message: "",
            LastHeartbeatTime: {
                Time: 0001-01-01T00:00:00Z,
            },
            LastTransitionTime: {
                Time: 0001-01-01T00:00:00Z,
            },
        },
    ]
to contain element matching
    <shared.Condition>: {
        Type: "Available",
        Status: "True",
        Reason: "",
        Message: "",
        LastHeartbeatTime: {
            Time: 0001-01-01T00:00:00Z,
        },
        LastTransitionTime: {
            Time: 0001-01-01T00:00:00Z,
        },
    }
/tmp/knmstate/kubernetes-nmstate/test/e2e/handler/conditions.go:134
```

**Comment by qinqon** [comment]
another run 
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/632/pull-kubernetes-nmstate-e2e-handler-k8s/1326895012364947456

**Comment by qinqon** [comment]
Now that we have the pod logs back we can see that the api-server probe is failing, that's why we don't see status at enactments since api-server cannot be reached enactments cannot be saved.

https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/651/pull-kubernetes-nmstate-e2e-handler-k8s/1333811072104140800/artifacts/NodeNetworkConfigurationPolicy%20bonding%20default%20interface%20when%20there%20is%20a%20default%20interface%20with%20dynamic%20address%20should%20successfully%20move%20default%20IP%20address%20on%20top%20of%20the%20bond_pods.log

```
   - eth0
    - eth1
  name: bond0
  state: up
  type: bond
 PolicyGeneration:1 Conditions:[{Type:Failing Status:True Reason:FailedToConfigure Message:error reconciling NodeNetworkConfigurationPolicy at desired state apply: , rolling back desired state configuration: failed runnig probes after network changes: failed runnig probe 'api-server' with after network reconfiguration -> currentState: ---
dns-resolver:
  config:
    search: []
    server: []
  running: {}
route-rules:
  config: []
routes:
 
```

**Comment by qinqon** [comment]
Maybe we can resolve it by having two masters so api-server get moved from one to the other by kubernetes, I don't if this can happend at kubevirtci.

**Comment by dhiller** [comment]
Happened again: 

https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/662/pull-kubernetes-nmstate-e2e-handler-k8s-future/1338879183568244736
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/662/pull-kubernetes-nmstate-e2e-handler-k8s/1338879183559856128

**Comment by qinqon** [comment]
Closed by #569

**Comment by alonSadan** [comment]
/reopen
happened  again: 
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/669/pull-kubernetes-nmstate-e2e-handler-k8s-future/1340990724841148416
